### DISCUSSION TITLE
More flexible frequency settings

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [iSoron/uhabits#813] [Feature Requests]
I have a recurring task that occurs 3 times per week, on Mon, Wed and Fri only. It should be 3 events (not 1) logged but I unfortunately there's no way to set repeating that way (i.e. weekly but with weekday mask). Can this be improved? I think that calendar app is good idea of how this should work:

![Screenshot_20210222-135953](https://user-images.githubusercontent.com/8041294/108712238-c2a11900-7516-11eb-86d7-36b67a10bcab.jpg)

![Screenshot_20210222-135940](https://user-images.githubusercontent.com/8041294/108712234-c16fec00-7516-11eb-915c-a8b98319c2bd.jpg)

![Screenshot_20210222-140041](https://user-images.githubusercontent.com/8041294/108712243-c339af80-7516-11eb-882b-0a01bb9db91c.jpg)

![screen](https://user-images.githubusercontent.com/8041294/108712584-465b0580-7517-11eb-9947-b14620210e75.png)

**Comment by iSoron** [comment]
> I have a recurring task that occurs 3 times per week, on Mon, Wed and Fri only. It should be 3 events (not 1) logged but I unfortunately there's no way to set repeating that way (i.e. weekly but with weekday mask).

Thank you for the feature request @MarcinOrlowski, but unfortunately this is not very clear to me. If you are creating one habit that should be done 3 times per week, then you can set the frequency to "3 times in 7 days". If you are creating three weekly habits, then you can set the frequency to "1 time in 7 days" for each one. Loop will automatically skip weekends in either case, after you have added enough repetitions in week days.

**Reply by MarcinOrlowski** [reply]
The habit should be done 3 times per week but in exact days. Let say I want to do a thing on Monday then on Thursday and Friday. On these exact days. So I'd set a reminder, it should ask me on Mon, Thu and Fri respectively. But now, watching the main list of habits I am not sure this would work in your app as all habits are shown with daily "marks" no matter how frequent the habit is. Also, even I am using 2.0.0-alpha, each day can only have "check", "x" and "?". While in my case I'd need 4th state saying "N/A". So I got a feeling that this case may simply not be fitting here.

**Reply by iSoron** [reply]
> The habit should be done 3 times per week but in exact days.

Loop already supports this. From the FAQ:

> **How can I specify that a habit should be performed on certain days of the week?**
> You don't need to specify this. The app will figure it out by itself. You just need to specify how many times would you like to repeat the habit in a certain interval of time. For example, if you want to repeat a certain habit every Monday and Friday, simply specify that you want to repeat the habit 2 times per week, and then start practicing your habit. If, during your first week, you indeed perform your habit on Monday and Friday, the app will automatically add a checkmark on the following Saturday and Sunday. As long as you keep your routine, the checkmarks should appear automatically on the correct days.

**Reply by MarcinOrlowski** [reply]
But that still would be presented in "daily context", while I would rather prefer to see only these days that are related to the schedule - so I only want to see 3 days not 7. I think this is not doable here and I need different tool or write my own. Feel free to close it you confirm.

**Reply by iSoron** [reply]
> while I would rather prefer to see only these days that are related to the schedule

If you activate "hide completed", Loop will also hide habits that are not scheduled for that day of the week. For example, if you have a weekly habit and you completed it on Monday, "hide completed" will hide it until the following Monday. This also works for more complex schedules.
